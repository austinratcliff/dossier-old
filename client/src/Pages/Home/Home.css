.Home {
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: #111;
  color: white;
  font-family: 'IBM Plex Mono', monospace;
  height: 100vh;
  width: 100vw;
  z-index: 0;
}

.Home .name {
  font-size: calc(10px + 5vmin);
  margin: 0;
}

.Home .title {
  font-size: calc(10px + 2vmin);
  margin: 15px 0 0 0;
}

.Home .icon {
  position: absolute;
  cursor: pointer;
}

.Home .about {
  top: 0;
  margin-top: 25px;
}

.Home .blog {
  left: 0;
  margin-left: 25px;
}

.Home .contact {
  bottom: 0;
  margin-bottom: 25px;
}

.Home .projects {
  right: 0;
  margin-right: 25px;
}
