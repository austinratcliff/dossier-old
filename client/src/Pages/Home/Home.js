import React from 'react'
import Navbar from 'react-bootstrap/Navbar'
import Nav from 'react-bootstrap/Nav'
import './Home.css'

const Home = props => {
  return (
    <div className="Home">
      <Navbar expand='lg' fixed='top'>
        <Navbar.Toggle>
          <i className='fas fa-ellipsis-h' title='menu' />
        </Navbar.Toggle>
        <Navbar.Collapse>
          <Nav>
            <i
              className='fas fa-user fa-2x icon about'
              title='about'
              onClick={props.handleClick}
            />
            <i className='fas fa-align-left fa-2x icon blog'
              title='blog'
              onClick={props.handleClick}
            />
            <i
              className='fas fa-comment-dots fa-2x icon contact'
              title='contact'
              onClick={props.handleClick}
            />
            <i
              className='fas fa-code fa-2x icon projects'
              title='projects'
              onClick={props.handleClick}
            />
          </Nav>
        </Navbar.Collapse>
      </Navbar>
      <h1 className="name">Austin Ratcliff</h1>
      <h2 className="title">Software Developer</h2>
    </div>
  )
}

export default Home
