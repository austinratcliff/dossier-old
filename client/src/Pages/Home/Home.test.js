import React from 'react'
import ReactDOM from 'react-dom'
import ReactTestRenderer from 'react-test-renderer'
import Home from './Home'

describe('Home component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Home />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('matches snapshot', () => {
    const component = ReactTestRenderer.create(<Home />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
