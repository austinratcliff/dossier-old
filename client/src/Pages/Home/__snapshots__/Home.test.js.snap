// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<div
  className="Home"
>
  <nav
    className="navbar navbar-expand-lg navbar-light fixed-top"
  >
    <button
      aria-label="Toggle navigation"
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-ellipsis-h"
        title="menu"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <i
          className="fas fa-user fa-2x icon about"
          title="about"
        />
        <i
          className="fas fa-align-left fa-2x icon blog"
          title="blog"
        />
        <i
          className="fas fa-comment-dots fa-2x icon contact"
          title="contact"
        />
        <i
          className="fas fa-code fa-2x icon projects"
          title="projects"
        />
      </div>
    </div>
  </nav>
  <h1
    className="name"
  >
    Austin Ratcliff
  </h1>
  <h2
    className="title"
  >
    Software Developer
  </h2>
</div>
`;
